src/tests/ftest/control/dmg_telemetry_net.py

```diff
@@ -0,0 +1,154 @@
+"""
+  (C) Copyright 2018-2021 Intel Corporation.
+
+    SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+
+import threading
+import time
+
+from re import search
+from mdtest_test_base import MdtestBase
+from telemetry_test_base import TestWithTelemetry
+from general_utils import DaosTestError, pcmd
+from command_utils import CommandFailure
+from apricot import skipForTicket
+
+# pylint: disable=too-few-public-methods,too-many-ancestors
+# pylint: disable=attribute-defined-outside-init
+class TestWithTelemetryNet(MdtestBase, TestWithTelemetry):
+    """Rebuild test cases featuring mdtest.
+
+    This class contains tests for pool rebuild that feature I/O going on
+    during the rebuild using IOR.
+
+    :avocado: recursive
+    """
+
+    def pre_tear_down(self):
+        """Tear down any test-specific steps prior to running tearDown().
+
+        Returns:
+            list: a list of error strings to report after all tear down
+            steps have been attempted
+
+        """
+        # kill any hanging mdtest processes
+        cmd = "pkill -9 mdtest"
+        for retry in range(0, 5):
+            ret_codes = pcmd(self.hostlist_servers, cmd)
+            if len(ret_codes) > 1 or 0 not in ret_codes:
+                self.log.info("Retry #%d: Not all '%s' commands in pcmd "
+                              "returned 0.", retry, cmd)
+
+        # Ignore errors cleaning up mdtest
+        return []
+
+    def tearDown(self):
+        try:
+            super().tearDown()
+        except (CommandFailure, DaosTestError) as excep:
+            self.log.info("%s: tearDown threw an exception "
+                          "because a rank was (intentionally) "
+                          "stopped midway through a run of mdtest.",
+                          repr(excep))
+
+    @skipForTicket("DAOS-9106")
+    def test_net_telemetry(self):
+        """Jira ID: DAOS-9020.
+
+        Test Description: Verify engine net telemetry metrics.
+
+        Use Cases:
+          * Create pool and container.
+          * Use mdtest to create 120K files with 3-way replication.
+          * Stop one server.
+          * Ensure engine_net_ofi_sockets_req_timeout is greater than 0.
+
+        :avocado: tags=all,pr,daily_regression
+        :avocado: tags=hw,large
+        :avocado: tags=control,telemetry,net
+        :avocado: tags=test_with_telemetry_net,test_net_telemetry
+        """
+
+        # Get the req_timeout metric name, which depends on the libfabric
+        # provider name
+        all_metrics = self.telemetry.get_all_server_metrics_names(
+            self.server_managers[0])
+        p = r"engine_net_\w+_req_timeout"
+        metrics = list(filter(lambda x: search(p, x), all_metrics))
+
+        # Initialize req_timeouts based on what's already populated in the
+        # telemtry system, which may or may not be zero.
+        req_timeouts_start = 0
+
+        data = {}
+        try:
+            data = self.telemetry.get_metrics(metrics[0])
```

Comment:
Without any sorting it seems like `metrics[0]` could possibly not end up being the metric name that is intended to be verified.  I feel it would be better to get all of the `metrics` values - this could even be done with a new `TelemetryUtils.get_engine_net_metrics(self, server_manager)`, negating the need for lines 76-80 - before running mdtest and after stopping a rank.  The two dictionaries could then be compared with something like `TelemetryUtils.verify_metric_values()` - where after reporting all the values/comparisons (for easier triage) a simple status could be returned to indicate if there were any values in the after list that were not greater than the before list.